src/interfaces/upsert-session.ts

```diff
@@ -0,0 +1,6 @@
+import { TUpsertOptions } from '../types/upsert-options';
+import { ISession } from './session';
+
+export interface IUpsertSession<W extends {}, R extends W, C extends {}, O extends {} = {}> extends ISession<W, R, C, TUpsertOptions<R, C> & O> {
+
```

Comment:
We should expose the `created` flag on the session, ie via `created()` and `updated()` methods, so that hooks know what to do.